npm/private/utils.bzl

```diff
@@ -156,6 +182,110 @@ def _npm_registry_download_url(package, version, registries, default_registry):
 def _is_git_repository_url(url):
     return url.startswith("git+ssh://")
 
+def _to_registry_url(url):
+    return "{}://{}".format(DEFAULT_REGISTRY_PROTOCOL, url) if url.find("//") == -1 else url
+
+def _default_registry():
+    return "{}://registry.npmjs.org/".format(DEFAULT_REGISTRY_PROTOCOL)
+
+def _hash(s):
+    result = str(hash(s))
+    if result.startswith("-"):
+        # Bazel's hash() resolves to a signed 32bit number [-2,147,483,648 to 2,147,483,647].
+        # Convert manually to a unique unsigned number here.
+        result = result[1:]
+        if len(result) <= 9:
+            padding = "0" * (9 - len(result))
+            result = "3" + padding + result
+        elif result[0] == "1":
+            result = "4" + result[1:]
+        elif result[0] == "2":
+            result = "5" + result[1:]
+        else:
+            fail(INTERNAL_ERROR_MSG)
+    return result
+
+def _dicts_match(a, b):
+    if len(a) != len(b):
+        return False
+    for key in a.keys():
+        if not key in b:
+            return False
+        if a[key] != b[key]:
+            return False
+    return True
+
+# Generate a consistent label string between Bazel versions.
+def _consistent_label_str(label):
+    return "//{}:{}".format(
+        # Starting in Bazel 6, the workspace name is empty for the local workspace and there's no other way to determine it.
+        # This behavior differs from Bazel 5 where the local workspace name was fully qualified in str(label).
+        label.package,
+        label.name,
+    )
+
+# Copies a file from the external repository to the same relative location in the source tree
+# TODO: test on Windows
```

Comment:
Since we have windows specific logic we should probably do this?

Reply:
yea, I'm going to dust off my windows box before landing

Reply:
can we drop the comment then?